RFC-005-SWAP-Bitcoin-basic.md

```diff
@@ -21,123 +21,154 @@
+## Motivation
 
-## Content
+The motivation for creating this RFC is to allow implementations to negotiate and execute [RFC003 Basic HTLC Atomic Swaps](./RFC-003-SWAP-basic.md) using the COMIT protocol. The identity definition introduced in this RFC should also be useful for future protocols.
 
-### Identities
+## Bitcoin Identity
 
-While it may seem obvious at first to use an address as an identity, such choice would come with a number of unnecessary manipulation and checks to validate the address. Such as verifying that it is a p2pkh or p2wpkh address, extraction of the public key hash, validation of the network.
+The Identity to be used on Bitcoin is the 20-byte *pubkeyhash* which is the result of applying SHA-256 and then RIPEMD-160 to a user's compressed SECP256k1 public key.
+The compressed public key is used because it needs to be compatiable with segwit transactions (see [BIP143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#Restrictions_on_public_key_type)).
 
-For simplicity purposes, the identity on bitcoin is defined as the public key hash. It is the same public key hash (SHA256+RIPEMD160) that is used to construct addresses, ensuring similar security level to sharing an address.
+While it may seem more intuitive to use a Bitcoin *address* as the identity, the pubkeyhash better fits the definition of identity given in RFC003. Using a Bitcoin address as the identity would require implementations to do a number of cumbersome validation steps such as such as verifying that it is a p2pkh or p2wpkh address, extraction of the pubkeyhash, validation of the network.
 
-Note that only compressed public keys are accepted as it needs to be usable in P2WSH[1].
+When using the pubkeyhash as one of the identities in an RFC003 message body it MUST be encoded as 20 hex encoded string.
+This RFC extends the [registry](./registry.md) with the following entry in the identity table:
 
 | Ledger   | Identity Name | JSON Encoding            | Description                                                                                  |
 |:----     |:-------       |:-------------            | -------------------------------------------------------------------------------------------- |
 | Bitcoin  | `pubkeyhash`  | `hex-encoded-bytes (20)` | The result of applying SHA256 and then RIPEMD160 to a user's SECP256k1 compressed public key |
 
-### Scripts
+## Hash Time Lock Contract
 
-#### Variables
+### Hash Functions
 
-The information in the table is needed to build the HTLC.
 
-the `[20]` denotation is used to express a push of the value between bracket onto the stack. Note that, as per usual Bitcoin script foramt, all values and placeholder presents MUST be encoded in hexadecimal without a leading `0x`. E.g. `[20]` would be replaced with `01 20` to push the decimal number `32` on the stack.
+The `hash_function` header must be set to one of the following values if Bitcoin is used as a ledger:
```

Comment:
Yes 🤯 but it's NYI

Reply:
Also note I was thinking about adding HASH160 as an alternative (so BTC to LTC could be done using that instead). It could be done on Ethereum as well but that would require two contract calls so I'd give it a miss.